cpp/src/worker.cpp

```diff
@@ -374,4 +428,16 @@ std::shared_ptr<Listener> Worker::createListener(uint16_t port,
   return listener;
 }
 
+void Worker::registerAmAllocator(ucs_memory_type_t memoryType, AmAllocatorType allocator)
+{
+  if (_amData == nullptr)
+    throw std::runtime_error("Active Messages wasn not enabled during context creation");
+  _amData->_allocators.insert_or_assign(memoryType, allocator);
```

Comment:
FWIW, since `ucs_memory_type_t` is an enum, the `_allocators` object could just be a `std::array<AmAllocatorType, UCS_MEMORY_TYPE_LAST + 1>` and accessing it could be `_allocators[memoryType]`.

Reply:
Given UCX leaves this for the user to write any handler, the result type could be anything, even something that UCX doesn't support. One could receive on a CUDA buffer and move that data to a host buffer before returning it to the user, for example. Another possibility would be some kind of composition that returns multiple buffers (we may want to explore a multi-buffer transfer perhaps, similar to `RequestTagMulti`). In any of those cases we will probably want a non-`ucs_memory_type_t` assuming we make this more flexible in the future and allow the user to do any arbitrary transfers.